Microsoft.Azure.Cosmos/src/Encryption/EncryptionSerializer.cs

```diff
@@ -0,0 +1,171 @@
+                    {
+                        propertyName = ((JsonPropertyAttribute)jsonProperty).PropertyName;
+                    }
+
+                    JToken propertyValueHolder = itemJObj.Property(propertyName).Value;
+
+                    // Even null in the JSON is a JToken with Type Null, this null check is just a sanity check
+                    if (propertyValueHolder != null)
+                    {
+                        toEncryptJObj.Add(propertyName, itemJObj.Property(propertyName).Value.Value<JToken>());
+                        itemJObj.Remove(propertyName);
+                    }
+                }
+            }
+
+            if (!toEncryptJObj.HasValues)
+            {
+                return this.baseSerializer.ToStream(input);
+            }
+
+            byte[] plainText = Encoding.UTF8.GetBytes(toEncryptJObj.ToString(Formatting.None));
+
+            Debug.Assert(inMemoryRawDek.AlgorithmUsingRawDek.AlgorithmName == AeadAes256CbcHmac256Algorithm.AlgorithmNameConstant);
+            EncryptionProperties encryptionProperties = new EncryptionProperties(
+                dataEncryptionKeyRid: dekProperties.ResourceId,
+                encryptionFormatVersion: 1,
+                encryptionAlgorithmId: 1,
+                encryptedData: inMemoryRawDek.AlgorithmUsingRawDek.EncryptData(plainText));
+
+            itemJObj.Add(Constants.Properties.EncryptedInfo, JObject.FromObject(encryptionProperties));
+            return this.baseSerializer.ToStream(itemJObj);
+        }
+
+        /// <inheritdoc />
+        public override async Task<T> FromStreamAsync<T>(Stream stream, Container container, ItemRequestOptions itemRequestOptions, CancellationToken cancellationToken)
+        {
+            JObject itemJObj = null;
+            using (StreamReader streamReader = new StreamReader(stream))
+            using (JsonTextReader jsonTextReader = new JsonTextReader(streamReader))
+            {
+                itemJObj = JObject.Load(jsonTextReader);
+            }
+
+            JProperty encryptionPropertiesJProp = itemJObj.Property(Constants.Properties.EncryptedInfo);
+            JObject encryptionPropertiesJObj = null;
+            if (encryptionPropertiesJProp != null && encryptionPropertiesJProp.Value != null && encryptionPropertiesJProp.Value.Type == JTokenType.Object)
+            {
+                encryptionPropertiesJObj = (JObject)encryptionPropertiesJProp.Value;
+            }
+
+            if (encryptionPropertiesJObj == null)
+            {
+                return itemJObj.ToObject<T>();
+            }
+
+            EncryptionProperties encryptionProperties = encryptionPropertiesJObj.ToObject<EncryptionProperties>();
+            if (encryptionProperties.EncryptionFormatVersion != 1)
+            {
+                throw new CosmosException(HttpStatusCode.InternalServerError, "Unknown encryption format version. Please upgrade your SDK to the latest version.");
```

Comment:
Please add the version to the error message.
```suggestion
                throw new CosmosException(HttpStatusCode.InternalServerError, $"Unknown encryption format version: {encryptionProperties.EncryptionFormatVersion}. Please upgrade your SDK to the latest version.");
```